pkg/dataobj/internal/dataset/row_range.go

```diff
@@ -0,0 +1,56 @@
+package dataset
+
+import (
+	"fmt"
+)
+
+type rowRange struct {
```

Comment:
nit: document inclusivity
```suggestion
// [start, end] inclusive
type rowRange struct {
```